odoo/addons/base/models/image_placeholder_mixin.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api
+from odoo.tools import generate_avatar_svg
+
+
+class ImagePlaceholderMixin(models.AbstractModel):
+    _name = 'image_placeholder.mixin'
+    _description = "Image Placeholder Mixin"
+
+    _name_field = "name"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    image_1920 = fields.Image("Image", max_width=1920, max_height=1920, compute="_compute_avatars", inverse=lambda partner: partner._set_avatar(1920), readonly=False)
+
+    image_1024 = fields.Image("Image 1024", related="image_1920", max_width=1024, max_height=1024, compute="_compute_avatars", inverse=lambda partner: partner._set_avatar(1024), readonly=False)
+    image_512 = fields.Image("Image 512", related="image_1920", max_width=512, max_height=512, compute="_compute_avatars", inverse=lambda partner: partner._set_avatar(512), readonly=False)
+    image_256 = fields.Image("Image 256", related="image_1920", max_width=256, max_height=256, compute="_compute_avatars", inverse=lambda partner: partner._set_avatar(256), readonly=False)
+    image_128 = fields.Image("Image 128", related="image_1920", max_width=128, max_height=128, compute="_compute_avatars", inverse=lambda partner: partner._set_avatar(128), readonly=False)
+
+    # image placeholder fields are stored and hold the image data
+
+    image_placeholder_1920 = fields.Image("Image Placeholder", max_width=1920, max_height=1920, store=True)
+
+    # resized fields stored for performance
+    image_placeholder_1024 = fields.Image("Image Placeholder 1024", related="image_placeholder_1920", max_width=1024, max_height=1024, store=True)
+    image_placeholder_512 = fields.Image("Image Placeholder 512", related="image_placeholder_1920", max_width=512, max_height=512, store=True)
+    image_placeholder_256 = fields.Image("Image Placeholder 256", related="image_placeholder_1920", max_width=256, max_height=256, store=True)
+    image_placeholder_128 = fields.Image("Image Placeholder 128", related="image_placeholder_1920", max_width=128, max_height=128, store=True)
```

Comment:
name to brainstorm, but this is wrong (it says it's placeholder but in reality it's the one without placeholder)

maybe image_store_1920